pyomo/contrib/fbbt/fbbt.py

```diff
@@ -443,10 +443,11 @@ def _prop_bnds_leaf_to_root_GeneralExpression(node, bnds_dict, feasibility_tol):
         region is removed due to floating point arithmetic and to prevent math domain errors (a larger value
         is more conservative).
     """
-    if node.expr.__class__ in native_types:
-        expr_lb = expr_ub = node.expr
+    (expr,) = node.args
+    if expr.__class__ in native_types:
+        expr_lb = expr_ub = expr
     else:
```

Comment:
Why did this need to change? What is wrong with using `expr`?

Reply:
`.expr` used to be an alias for `.arg(0)` and `.args[0]`.  This PR is changing that so that `expr` is guaranteed to return a `PyomoObject`.  However, it is not guaranteed to return the *same* `PyomoObject` each time.  Because FBBT caches the `id()` of all `PyomoObject`s, the easiest (and correct) fix is to retrieve the raw `.args` and not an expression.